Controller/MediaAdminController.php

```diff
@@ -108,4 +115,101 @@ public function listAction(Request $request = null)
+
+            $parameters = $this->admin->getPersistentParameters();
+
+            $provider = $this->getRequest()->get('provider');
+
+
+        if(!$provider) {
+            $providers = $this->get('sonata.media.pool')->getProvidersByContext($this->get('request')->get('context', $this->get('sonata.media.pool')->getDefaultContext()));
+
+            $filteredProviders = array();
+            foreach ($providers as $key => $provider) {
+                if ($provider instanceof FileProvider) {
+                    $filteredProviders[] = $provider;
+                }
+            }
+
+            return $this->render('SonataMediaBundle:MediaAdmin:select_provider.html.twig', array(
+                'providers'     => $filteredProviders,
+                'base_template' => $this->getBaseTemplate(),
+                'admin'         => $this->admin,
+                'action'        => 'multi_upload',
+            ));
+
+        }
+
+        $form = $this->createForm(new MultiUploadType(), null , array('provider' => $provider, 'context' => $parameters['context']));
+
+        return $this->render('SonataMediaBundle:MediaAdmin:multi_upload.html.twig', array(
+            'action' => 'multi_upload',
+            'multi_form' => $form->createView()
+        ));
+    }
+
+    public function multiUploadAjaxAction(Request $request)
+    {
+        if (false === $this->admin->isGranted('CREATE')) {
+            throw new AccessDeniedHttpException();
+        }
+
+        $form = $this->createForm(new MultiUploadType());
+        $form->handleRequest($request);
+
+        $logger = $this->get('logger');
+
+        $files = array();
+        if ($form->isValid()) {
+            $data = $form->getData();
+
+            $uploadedFiles = $data['files'];
+            $mediaPool = $this->get('sonata.media.pool');
+            /** @var $uploadedFile UploadedFile */
+            foreach ($uploadedFiles as $uploadedFile) {
+                if ($uploadedFile->isValid()) {
+                    $this->admin->setRequest($request);
+                    /** @var $media MediaInterface */
+                    $media = $this->admin->getNewInstance();
+                    $media->setProviderName($data['provider']);
+                    $media->setContext($data['context']);
+                    $media->setBinaryContent($uploadedFile);
+
+                    try {
+                        $media = $this->admin->update($media);
+                        $provider = $mediaPool->getProvider($data['provider']);
+
+                        $files[] = array(
+                            'name' => $media->getName(),
+                            'editUrl' => $this->admin->generateUrl('edit', array('id' => $media->getId())),
+                            'thumbnailUrl' => $provider->getCdnPath($provider->getReferenceImage($media), true)
+                        );
+                    } catch(\Exception $e) {
+                        $logger->error("Fehler beim MultiUpload", array($e));
```

Comment:
Should be `array('exception' => $e)` (the `exception` is special, it gives a nicer display in Sentry, for instance).